package/harvester-os/files/usr/sbin/migration/migrate_harv_mgmt_to_mgmt_br.sh

```diff
@@ -0,0 +1,161 @@
+	yq -i 'del( .stages.initramfs[0].files[] | select(.path == "*ifroute*"))' "$HARV_CONFIG"
+	sed -i "s/iface: $HARV_MGMT/iface: \"\"/g" "$HARV_CONFIG"
+
+	# add file
+	if [ $MODE == "dhcp" ]; then
+		yq -i '.stages.initramfs[0].commands += "rm -f /etc/sysconfig/network/ifroute-mgmt-br"' "$HARV_CONFIG"
+	fi
+
+	cat <<EOF | yq eval-all -i 'select(fileIndex==0).stages.initramfs[0].files += select(fileIndex==1) | select(fileIndex==0)' "$HARV_CONFIG" -
+- path: /etc/wicked/scripts/setup_bond.sh
+  permissions: 755
+  owner: 0
+  group: 0
+  content: |+
+    #!/bin/sh
+
+    ACTION=\$1
+    INTERFACE=\$2
+
+    case \$ACTION in
+    	post-up)
+    		# accept all vlan, PVID=1 by default
+    		bridge vlan add vid 2-4094 dev \$INTERFACE
+    		;;
+    esac
+  encoding: ""
+  ownerstring: ""
+
+- path: /etc/wicked/scripts/setup_bridge.sh
+  permissions: 755
+  owner: 0
+  group: 0
+  content: |+
+    #!/bin/sh
+
+    ACTION=\$1
+    INTERFACE=\$2
+
+    case \$ACTION in
+    	pre-up)
+    		# enable vlan-aware
+    		ip link set \$INTERFACE type bridge vlan_filtering 1
+    		;;
+
+    	post-up)
+    		# inherit MAC address from mgmt-bo
+    		ip link set dev \$INTERFACE address \$(ip -json link show dev mgmt-bo | jq -j '.[0]["address"]')
+
+    		# accept all vlan, PVID=1 by default
+    		bridge vlan add vid 2-4094 dev \$INTERFACE self
+    		bridge vlan add vid 2-4094 dev mgmt-bo
+    		;;
+    esac
+  encoding: ""
+  ownerstring: ""
+
+- path: /etc/sysconfig/network/ifcfg-mgmt-br
+  permissions: 384
+  owner: 0
+  group: 0
+  content: |+
+    STARTMODE='onboot'
+    BOOTPROTO='${MODE}'
+    BRIDGE='yes'
+    BRIDGE_STP='off'
+    BRIDGE_FORWARDDELAY='0'
+    BRIDGE_PORTS='mgmt-bo'
+    PRE_UP_SCRIPT="wicked:setup_bridge.sh"
+    POST_UP_SCRIPT="wicked:setup_bridge.sh"
+
+    $(echo "$IFCFG" | grep "IPADDR")
+    $(echo "$IFCFG" | grep "NETMASK")
+    $(echo "$IFCFG" | grep "DHCLIENT_SET_DEFAULT_ROUTE")
+    $(echo "$IFCFG" | grep "MTU")
+  encoding: ""
+  ownerstring: ""
+
+- path: /etc/sysconfig/network/ifcfg-mgmt-bo
+  permissions: 384
+  owner: 0
+  group: 0
+  content: |+
+    STARTMODE='onboot'
+    BONDING_MASTER='yes'
+    BOOTPROTO='none'
+    POST_UP_SCRIPT="wicked:setup_bond.sh"
+    $(echo "$IFCFG" | grep "BONDING_SLAVE_")
+    $(echo "$IFCFG" | grep "BONDING_MODULE_OPTS")
+
+    $(echo "$IFCFG" | grep "MTU")
+  encoding: ""
+  ownerstring: ""
+EOF
+
+	if [ $MODE == "static" ]; then
+		cat <<EOF | yq eval-all -i 'select(fileIndex==0).stages.initramfs[0].files += select(fileIndex==1) | select(fileIndex==0)' "$HARV_CONFIG" -
+- path: /etc/sysconfig/network/ifroute-mgmt-br
+  permissions: 384
+  owner: 0
+  group: 0
+  content: |
+    $(echo "$IFROUTE" | sed "s/$HARV_MGMT/mgmt-br/g")
+  encoding: ""
+  ownerstring: ""
+EOF
+	fi
+}
+
+if ! grep -q "$HARV_MGMT" "$HARV_CONFIG"; then
+	echo "$HARV_MGMT not found. Skipping."
+	exit 0
+fi
+
+echo "$HARV_MGMT found."
+
+# check DHCP or static
+
+if grep -q "BOOTPROTO='static'" "$HARV_CONFIG"; then
```

Comment:
The detection needs to be in `harvester-mgmt` file block only. Since the user might create more bonding devices and their `file` block also can contain the `BOOTPROTO=xxx` string.

Reply:
fixed. only detect harvester-mgmt's type now.